### PR TITLE
Pin PyYAML version compatible with Python 3.8+

### LINKED ISSUE #77927: AttributeError: module 'collections' has no attribute 'Hashable'
### Summary

It seems an upgrade broke my ansible configuration on Fedora 36 not much to say any commands is followed by the output past down in the issue.
I have seen this error, but for another package "Spades", don't know what it is, don't have installed on my machine : 
- https://github.com/ablab/spades/issues/873
- https://github.com/ablab/spades/issues/863
- https://github.com/pyinvoke/invoke/pull/803

Kernel: 5.17.7-300.fc36.x86_64  

Best regards.

### Issue Type

Bug Report

### Component Name

dnf

### Ansible Version

```console
$ ansible rpm -qa | grep ansible

ansible-core-2.12.5-1.fc36.noarch
ansible-5.8.0-1.fc36.noarch

```

### Configuration

```console
$ ansible-config dump --only-changed -t all

Traceback (most recent call last):
  File "/usr/bin/ansible-config", line 65, in <module>
    import ansible.constants as C
  File "/usr/lib/python3.10/site-packages/ansible/constants.py", line 180, in <module>
    config = ConfigManager()
  File "/usr/lib/python3.10/site-packages/ansible/config/manager.py", line 291, in __init__
    self._base_defs = self._read_config_yaml_file(defs_file or ('%s/base.yml' % os.path.dirname(__file__)))
  File "/usr/lib/python3.10/site-packages/ansible/config/manager.py", line 312, in _read_config_yaml_file
    return yaml_load(config_def) or {}
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/__init__.py", line 72, in load
    return loader.get_single_data()
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/constructor.py", line 37, in get_single_data
    return self.construct_document(node)
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/constructor.py", line 46, in construct_document
    for dummy in generator:
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/constructor.py", line 398, in construct_yaml_map
    value = self.construct_mapping(node)
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/constructor.py", line 204, in construct_mapping
    return super().construct_mapping(node, deep=deep)
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/constructor.py", line 126, in construct_mapping
    if not isinstance(key, collections.Hashable):
AttributeError: module 'collections' has no attribute 'Hashable'
```

### OS / Environment

 OS: Fedora release 36 (Thirty Six) x86_64 

### Expected Results

Any ansible commands return the same output bellow.

### Actual Results

```console
Traceback (most recent call last):
  File "/usr/bin/ansible-config", line 65, in <module>
    import ansible.constants as C
  File "/usr/lib/python3.10/site-packages/ansible/constants.py", line 180, in <module>
    config = ConfigManager()
  File "/usr/lib/python3.10/site-packages/ansible/config/manager.py", line 291, in __init__
    self._base_defs = self._read_config_yaml_file(defs_file or ('%s/base.yml' % os.path.dirname(__file__)))
  File "/usr/lib/python3.10/site-packages/ansible/config/manager.py", line 312, in _read_config_yaml_file
    return yaml_load(config_def) or {}
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/__init__.py", line 72, in load
    return loader.get_single_data()
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/constructor.py", line 37, in get_single_data
    return self.construct_document(node)
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/constructor.py", line 46, in construct_document
    for dummy in generator:
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/constructor.py", line 398, in construct_yaml_map
    value = self.construct_mapping(node)
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/constructor.py", line 204, in construct_mapping
    return super().construct_mapping(node, deep=deep)
  File "/home/paco.garcia/.local/lib/python3.10/site-packages/yaml/constructor.py", line 126, in construct_mapping
    if not isinstance(key, collections.Hashable):
AttributeError: module 'collections' has no attribute 'Hashable'
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Fixes #77927

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # packages, not optional ones, and with the widest range of versions that could
 # be suitable)
 jinja2 >= 3.0.0
-PyYAML
+PyYAML >= 5.1
```
Comment (mattclay) [inline requirements.txt]:
```suggestion
PyYAML >= 5.1  # PyYAML 5.1 is required for Python 3.8+ support
```

File: test/lib/ansible_test/_data/requirements/ansible.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@
 # packages, not optional ones, and with the widest range of versions that could
 # be suitable)
 jinja2 >= 3.0.0
-PyYAML
+PyYAML >= 5.1
```
Comment (mattclay) [inline test/lib/ansible_test/_data/requirements/ansible.txt]:
```suggestion
PyYAML >= 5.1  # PyYAML 5.1 is required for Python 3.8+ support
```

**Review by mattclay — CHANGES_REQUESTED** [review]
This should have a changelog fragment.

**Review by bcoca — APPROVED** [review]

File: changelogs/fragments/77936-add-pyyaml-version.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - Add PyYAML >= 5.1 as a dependecy of ansible-core to be compatible with Python 3.8+.
```
Comment (mattclay) [inline changelogs/fragments/77936-add-pyyaml-version.yml]:
```suggestion
  - Add PyYAML >= 5.1 as a dependency of ansible-core to be compatible with Python 3.8+.
```

**Review by mattclay — CHANGES_REQUESTED** [review]

File: changelogs/fragments/77936-add-pyyaml-version.yml (hunk 3 above)
Comment (s-hertel) [reply changelogs/fragments/77936-add-pyyaml-version.yml]: Just fixed :) Thanks

**Review by mattclay — APPROVED** [review]

**Pull request merged**